pandora/models/pandora.py

```diff
@@ -284,37 +284,99 @@ class BookmarkList(PandoraModel):
 
 class SearchResultItem(PandoraModel):
 
-    artist = Field("artistName")
-    song_name = Field("songName")
     score = Field("score")
-    likely_match = Field("likelyMatch", default=False)
     token = Field("musicToken")
 
     @property
     def is_song(self):
-        return self.token.startswith('S')
+        return type(self) is SongSearchResultItem
```

Comment:
This should really use 'isinstance' instead of 'type'. Same for the other checks.